Sources/Penny/BotStateManager.swift

```diff
@@ -2,64 +2,123 @@ import DiscordBM
+   * When `initialize()` is called, this will send a "please shutdown" message.
+   * When the old Penny instance receive that message.
+     * The old instance will cache save all needed cached stuff into a S3 bucket.
+     * Then it will send a "did shutdown" message.
+   * The new instance will catch the "did shutdown" message.
+     * The new instance will get the stuff from the S3 bucket.
+     * Then it will start allowing Gateway-event handlings.
+   * If the old instance is too slow to make the process happen, the process is aborted and
+     the new instance will start handling events without waiting more for the old instance.
  */
 actor BotStateManager {
     
     var logger = Logger(label: "BotStateManager")
-    var canRespond = true
+    var canRespond = false
     let id = Int(Date().timeIntervalSince1970)
-    
-    let signal = "Hello the other Pennys 👋 you can retire now :)"
+
+    var cachesService: any CachesService {
+        ServiceFactory.makeCachesService()
+    }
+
     var disableDuration = Duration.seconds(3 * 60)
-    
+
+    var isCachePopulated = false
+
     static private(set) var shared = BotStateManager()
     
     private init() { }
     
-    func initialize() {
+    func initialize() async {
         self.logger[metadataKey: "id"] = "\(self.id)"
-        Task { await sendSignal() }
+        Task { await send(.shutdown) }
+        cancelIfCachePopulationTakesTooLong()
     }
-    
+
+    func cancelIfCachePopulationTakesTooLong() {
+        Task {
+            try await Task.sleep(for: .seconds(15))
+            if !isCachePopulated {
+                canRespond = true
+                logger.error("No CacheStorage-population signal was done in-time")
+            }
+        }
+    }
+
     func canRespond(to event: Gateway.Event) -> Bool {
-        checkIfItsASlowdownSignal(event: event)
+        checkIfItsASignal(event: event)
         return canRespond
     }
     
-    private func checkIfItsASlowdownSignal(event: Gateway.Event) {
+    private func checkIfItsASignal(event: Gateway.Event) {
         guard case let .messageCreate(message) = event.data,
               message.channel_id == Constants.Channels.logs.id,
               let author = message.author,
-              author.id.rawValue == Constants.botId,
-              message.content.hasPrefix(signal)
+              author.id.rawValue == Constants.botId
         else { return }
         guard let otherId = message.content.split(whereSeparator: \.isWhitespace).last else {
             logger.warning("Can't find id of the other Penny")
             return
         }
         if otherId == "\(self.id)" { return }
+        if message.content.hasPrefix(StateManagerSignal.shutdown.value) {
+            shutdown()
+        } else if message.content.hasPrefix(StateManagerSignal.didShutdown.value) {
+            populateCache()
+        } else {
+            logger.error("Unknown signal")
+            return
```

Comment:
Nit: This trick (also, say what the signal was...)
```suggestion
            return logger.error("Unknown signal", metadata: ["signal": .string(message.content)])
```